runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/PartitioningShuffleReader.java

```diff
@@ -144,9 +143,10 @@ public boolean advance() throws IOException {
         return false;
       }
       ShuffleEntry record = iterator.next();
-      K key = CoderUtils.decodeFromByteArray(shuffleReader.keyCoder, record.getKey());
+      K key = shuffleReader.keyCoder.decode(record.getKey().newInput(), Coder.Context.OUTER);
```

Comment:
This check is that all the bytes are read is important as it catchers coder issues:
https://github.com/apache/beam/blob/e62ae391985fc13c7df1ee6e088525835ceaa560/sdks/java/core/src/main/java/org/apache/beam/sdk/util/CoderUtils.java#L102

Please re-add the check here and elsewhere (probably via a convenience method of some kind)